src/crimson/os/seastore/transaction.h

```diff
@@ -83,10 +86,15 @@ class Transaction {
     ceph_assert(inserted);
   }
 
-  void add_fresh_extent(CachedExtentRef ref) {
+  void add_fresh_extent(CachedExtentRef ref,
+			std::optional<paddr_t> zero_paddr = std::nullopt) {
     ceph_assert(!is_weak());
     fresh_block_list.push_back(ref);
-    ref->set_paddr(make_record_relative_paddr(offset));
+    if (zero_paddr) {
+      ref->set_paddr(*zero_paddr);
+    } else {
+      ref->set_paddr(make_record_relative_paddr(offset));
+    }
     offset += ref->get_length();
```

Comment:
It is not correct to increment the `offset` (the record relative paddr) if it is an ool extent.

Reply:
yes, that's right.